stdlib/test/builtin/test_string.mojo

```diff
@@ -223,6 +223,47 @@ fn test_stringref_from_dtypepointer() raises:
     assert_equal(a, b)
 
 
+fn test_stringref_split() raises:
```

Comment:
**Suggestion** Should we put this test in `stdlib/test/builtin/test_stringref.mojo` instead?